doc/_scripts/software_maturity/software_maturity_scanner.py

```diff
@@ -295,7 +295,7 @@ def find_all_socs(all_samples: List[str]) -> List[str]:
 
     # Only use nRF boards, but not nRF51 series
     all_boards = set(filter(BOARD_FILTER, all_boards))
-    all_socs = {re.findall(r"_(nrf[0-9]+)", b)[-1] for b in all_boards}
+    all_socs = {re.findall(r"_(/nrf[0-9]+[hl]?[0-9]*/i)", b)[-1] for b in all_boards}
```

Comment:
```suggestion
    all_socs = {re.findall(r"_(nrf[0-9]+[hlHL]?[0-9]+)", b)[-1] for b in all_boards}
```

Reply:
This regexp will work as expected:
![image](https://github.com/nrfconnect/sdk-nrf/assets/85483674/db188e7d-de5d-4f58-9fd6-44d5cf726d99)

Reply:
generated yaml indicated that if needed, it must be done in different place.
![image](https://github.com/nrfconnect/sdk-nrf/assets/85483674/46017ba1-4c7c-4c46-b2be-15894c0c413a)
So, I would suggest, to apply the change and we will see how it works

Reply:
I found in https://github.com/nrfconnect/sdk-nrf/blob/main/doc/_extensions/software_maturity_table.py 
```
        for soc in all_socs:
            soc = soc.upper().replace("NRF", "nRF")
            entry = nodes.entry()
            entry += nodes.paragraph(text=soc)
            row += entry
        thead.append(row)
```
So, in documentation, the names should look OK